### PR TITLE
OpenModelica 1.17.0 package

### PR DESCRIPTION
**Pull request by balodja**
###### Motivation for this change

OpenModelica package has been broken for quite a long time. Let's bring it back into living. Aiming to finish #110588 and #54383 here.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122828` at 66295804 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/f83e488110d6e192921e111bcfc61c2f">openmodelica.omcompiler</a></li>
    <li><a href="https://gist.github.com/92d6f9e63d365cce37af39d07cd04b2e">openmodelica.omedit</a></li>
    <li><a href="https://gist.github.com/46f23b6b3c58e37f0f0b4380b7a1eb3d">openmodelica.omlibrary</a></li>
    <li><a href="https://gist.github.com/abe6108e4aca369e8cdb284432493300">openmodelica.omparser</a></li>
    <li><a href="https://gist.github.com/f2843c962cbf131030920792688d1d0c">openmodelica.omplot</a></li>
    <li><a href="https://gist.github.com/e43a83c6b5e7477577b7817b0093d051">openmodelica.omsimulator</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by balodja** [comment]
@grwlf , Sergey, could you take a look please? I'm struggling at this point to move any further as I'm not confident enough that my work would be accepted.

File: pkgs/applications/science/misc/openmodelica/omsimulator/default.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{pkg-config, readline, libxml2,
+openmodelica, mkOpenModelicaDerivation}:
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omsimulator/default.nix:2]:
```suggestion
{ pkg-config, readline, libxml2
, openmodelica, mkOpenModelicaDerivation
}:
```

File: pkgs/applications/science/misc/openmodelica/omsimulator/default.nix [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+mkOpenModelicaDerivation rec {
+  pname = "omsimulator";
+  omdir = "OMSimulator";
+  omdeps = [openmodelica.omcompiler];
+
+  nativeBuildInputs = [pkg-config];
+
+  buildInputs = [readline libxml2];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omsimulator/default.nix:11]:
```suggestion
  omdeps = [ openmodelica.omcompiler ];

  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ readline libxml2 ];
```

File: pkgs/applications/science/misc/openmodelica/omshell/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    sed -i OMShell/mosh/src/Makefile.in -e '
+      /^CFLAGS =/ s|-I../../../build|-I${omcompiler}|
+      /^LIBS =/ s|-L@OMBUILDDIR@|-L${omcompiler}|
+      '
+    '';
+
+  dontUseQmakeConfigure = true;
+
+#  NIX_CFLAGS_COMPILE = "-isystem ${omcompiler}/include/omc/c";
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omshell/default.nix:34]:
```suggestion
}
```

File: pkgs/applications/science/misc/openmodelica/omshell/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{lib, qttools, qmake, qtwebkit, wrapQtAppsHook, readline, openmodelica, mkOpenModelicaDerivation}:
+
+with openmodelica;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omshell/default.nix:3]: Please only apply with where it is required and not over the entire file.

File: pkgs/applications/science/misc/openmodelica/omplot/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+mkOpenModelicaDerivation rec {
+  pname = "omplot";
+  omdir = "OMPlot";
+  omdeps = [openmodelica.omcompiler];
+  omautoconf = true;
+
+  nativeBuildInputs = [qtbase qttools qmake wrapQtAppsHook];
+
+  buildInputs = [];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omplot/default.nix:12]:
```suggestion
```

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+CWD=`pwd`
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh:3]:
```suggestion
CWD=$PWD
```

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh:1]:
```suggestion
#!/usr/bin/env bash
```

/bin/sh is a symlink to bash on nixos and dash on Debian like systems which can cause problems.

Also this must be a nix-shell or most of the tool used below are not available.

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+T=`mktemp -d`
+trap "rm -rf $T" EXIT INT PIPE
+cd $T
+cat >check.nix <<EOF
+with import <nixpkgs> {};
+fetchgit `cat $CWD/../mkderivation/src-main.nix`
+EOF
+nix-build check.nix
+cat result/libraries/Makefile.libs
+) }
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh:15]:
```suggestion
chko() {
  (
    T=`mktemp -d`
    trap "rm -rf $T" EXIT INT PIPE
    cd $T
    cat >check.nix <<EOF
with import <nixpkgs> {};
fetchgit `cat $CWD/../mkderivation/src-main.nix`
    EOF
    nix-build check.nix
    cat result/libraries/Makefile.libs
  )
}
```

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+getsha256() { (
+URL=$(echo "$1" | sed 's/^"\(.*\)"$/\1/')
+REV=$(echo "$2" | sed 's/^"\(.*\)"$/\1/')
+SHA=$(nix run nixpkgs.nix-prefetch-git -c nix-prefetch-git --fetch-submodules "$URL" "$REV" 2>/dev/null | sed -n 's/.*"sha256": "\(.*\)",/\1/g p')
+echo "{ url = $1; rev = $2; sha256 = \"$SHA\"; fetchSubmodules = true; }"
+) }
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh:22]:
```suggestion
getsha256() {
  URL=$(echo "$1" | sed 's/^"\(.*\)"$/\1/')
  REV=$(echo "$2" | sed 's/^"\(.*\)"$/\1/')
  SHA=$(nix run nixpkgs.nix-prefetch-git -c nix-prefetch-git --fetch-submodules "$URL" "$REV" 2>/dev/null | sed -n 's/.*"sha256": "\(.*\)",/\1/g p')
  echo "{ url = $1; rev = $2; sha256 = \"$SHA\"; fetchSubmodules = true; }"
}
```

File: pkgs/applications/science/misc/openmodelica/omlibrary/default.nix [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+in
+mkOpenModelicaDerivation {
+  pname = "omlibrary";
+  omdir = "libraries";
+  omtarget = "omlibrary-all";
+  omdeps = [openmodelica.omcompiler];
+
+  nativeBuildInputs = [];
+
+  buildInputs = [];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/default.nix:14]:
```suggestion
```

File: pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix [hunk 11]
```diff
@@ -0,0 +1,7 @@
+{
+    url = "https://github.com/OpenModelica/OpenModelica/";
+    rev = "08fd3f9144235f209a4ed7602bfadb32b1823628";
+    sha256 = "0clgqk9ilnr43iyl5sdzwfzqpnw9amfy1npdgkpgm1wfnsvz6xrw";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix:5]:
```suggestion
  url = "https://github.com/OpenModelica/OpenModelica/";
  rev = "08fd3f9144235f209a4ed7602bfadb32b1823628";
  sha256 = "0clgqk9ilnr43iyl5sdzwfzqpnw9amfy1npdgkpgm1wfnsvz6xrw";
  fetchSubmodules = true;
```

Why is this not using fetchFromGitHub?

File: pkgs/applications/science/misc/openmodelica/mkderivation/default.nix [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+  inherit preAutoreconf;
+  inherit configureFlags configurePhase;
+  inherit preBuild;
+  inherit makeFlags installFlags;
+
+  src = fetchgit (import ./src-main.nix);
+  version = "1.17.0";
+
+  nativeBuildInputs = pkg.nativeBuildInputs ++
+    [autoconf automake libtool cmake autoreconfHook];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/mkderivation/default.nix:101]:
```suggestion
  nativeBuildInputs = pkg.nativeBuildInputs
    ++ [ autoconf automake libtool cmake autoreconfHook ];
```

File: pkgs/applications/science/misc/openmodelica/mkderivation/default.nix [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+    appendByAttr "installFlags" " " pkg;
+
+
+in stdenv.mkDerivation (pkg // {
+  inherit omtarget;
+  inherit postPatch;
+  inherit preAutoreconf;
+  inherit configureFlags configurePhase;
+  inherit preBuild;
+  inherit makeFlags installFlags;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/mkderivation/default.nix:95]:
```suggestion
  inherit omtarget postPatch preAutoreconf configureFlags configurePhase preBuild makeFlags installFlags;
```

File: pkgs/applications/science/misc/openmodelica/combined/default.nix [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+symlinkJoin {
+  name = "openmodelica-combined";
+  paths = [omcompiler omsimulator omplot omparser omedit omlibrary omshell];
+
+  buildInputs = [gnumake makeWrapper];
+
+  postBuild = ''
+    wrapProgram $out/bin/OMEdit \
+      --prefix PATH : "${gnumake}/bin" \
+      --prefix PATH : "${stdenv.cc}/bin" \
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/combined/default.nix:12]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ gnumake stdenv.cc ]} \
```

File: pkgs/applications/science/misc/openmodelica/combined/default.nix [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+      --prefix LIBRARY_PATH : "${lib.makeLibraryPath [blas lapack]}" \
+      --set-default OPENMODELICALIBRARY "${omlibrary}/lib/omlibrary"
+    '';
+}
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/combined/default.nix:17]: One final new line to much.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please format all files with nixpkgs-fmt.

Meta is missing everywhere.

File: pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix (hunk 11 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix:5]: fetchFromGitHub downloads zip's without submodules.

File: pkgs/applications/science/misc/openmodelica/omshell/default.nix (hunk 4 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/omshell/default.nix:3]: OK!

File: pkgs/applications/science/misc/openmodelica/combined/default.nix (hunk 15 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/combined/default.nix]: Done.

**Comment by balodja** [comment]
> Please format all files with nixpkgs-fmt.

Nice tool. Thank you for suggestion.

> Meta is missing everywhere.

Meta is one for all the OpenModelica-packages, it is located in mkderivation/default.nix. All packages are constructed with this helper.

**Comment by SuperSandro2000** [comment]
I don't have the time right now to look through so many files. Sorry.

**Comment by siraben** [comment]
I can help review some parts (will rely on nixpkgs-review for the rest), but it's good to see a package rise from a zombie state!

File: pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+      rpl = lib.replaceChars [ ":" "/" ] [ "_" "_" ];
+    in
+    (rpl r.url) + "-" + (rpl r.rev);
+
+in
+stdenv.mkDerivation {
+  name = "fakegit";
+
+  buildCommand = ''
+        mkdir -pv $out/repos
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix:25]: We indent with 2 spaces.

File: pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix [hunk 17]
```diff
@@ -0,0 +1,53 @@
+{ lib, stdenv, fetchgit, bash }:
+let
+  mkscript = path: text: ''
+    mkdir -pv `dirname ${path}`
+    cat > ${path} <<"EOF"
+    #!${bash}/bin/bash
+    ME=`basename ${path}`
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix:7]:
```suggestion
    ME=$(basename ${path})
```

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+
+chko() {
+  (
+    T=`mktemp -d`
+    trap "rm -rf $T" EXIT INT PIPE
+    cd $T
+    cat >check.nix <<EOF
+with import <nixpkgs> {};
+fetchgit `cat $CWD/../mkderivation/src-main.nix`
+    EOF
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh:13]:
```suggestion
EOF
```

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh [hunk 19]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh:1]: This should be a nix-shell.

File: pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix [hunk 20]
```diff
@@ -0,0 +1,7 @@
+{
+  url = "https://github.com/OpenModelica/OpenModelica/";
+  rev = "08fd3f9144235f209a4ed7602bfadb32b1823628";
+  sha256 = "0clgqk9ilnr43iyl5sdzwfzqpnw9amfy1npdgkpgm1wfnsvz6xrw";
+  fetchSubmodules = true;
+}
+# Update with: nix run -f ./nixpkgs/default.nix nix-prefetch-git -c nix-prefetch-git 'https://github.com/OpenModelica/OpenModelica/' 'v1.17.0' --fetch-submodules
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix:7]: Please move this into the derivation and use fetchFromGitHub.

File: pkgs/applications/science/misc/openmodelica/mkderivation/default.nix [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+    getAttrDef "configureFlags" [ ] pkg;
+
+  # Our own configurePhase that accounts for omautoconf
+  configurePhase = ''
+    runHook preConfigure
+    export configureFlags="''${configureFlags} --with-ombuilddir=$PWD/build --prefix=$prefix"
+    ./configure --no-recursion $configureFlags;
+    if ${if omautoconf then "true" else "false"}; then
+      (cd ${omdir}; ./configure $configureFlags);
+    fi
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/openmodelica/mkderivation/default.nix:65]:
```suggestion
    ${lib.optionalString omautoconf "(cd ${omdir}; ./configure $configureFlags)"}
```

**Review by SuperSandro2000 — COMMENTED** [review]
meta is missing in the files.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 122828` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix (hunk 16 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix]: OK, done in f1292a953e5

File: pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix (hunk 17 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/omlibrary/fakegit.nix]: Thanks. f1292a953e5

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh (hunk 18 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh]: No objections. f1292a953e5

File: pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh (hunk 19 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/omlibrary/update-src-libs.sh]: Ah, new thing to me. Fixed. f1292a953e5

File: pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix (hunk 20 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix:7]: Sorry, that is impossible for two reasons:
* `fetchFromGitHub` does not fetch submodules for OpenModelica. I have tried that earlier.
* We use `src-main.nix` in two places -- in `mkderivation/default.nix` and in `omlibrary/update-src-libs.sh`. So better that to be in a separate file.

File: pkgs/applications/science/misc/openmodelica/mkderivation/default.nix (hunk 21 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/mkderivation/default.nix]: Thanks, done. f1292a953e5

**Comment by balodja** [comment]
> meta is missing in the files.

OK, I see that placing meta in the single file in reusing it for several derivations is misleading. So I added separate metas for every derivation with their own personalized descriptions.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 122828` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by balodja** [comment]
> Result of `nixpkgs-review pr 122828` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

@siraben, it seems like something went wrong. I don't see any review details, or maybe I'm missing something important.

**Comment by siraben** [comment]
```
-bash-4.2$ nixpkgs-review pr --post-result 122828
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/122828/head:refs/nixpkgs-review/1
remote: Enumerating objects: 1294, done.
remote: Counting objects: 100% (940/940), done.
remote: Compressing objects: 100% (110/110), done.
remote: Total 1294 (delta 839), reused 900 (delta 821), pack-reused 354
Receiving objects: 100% (1294/1294), 950.90 KiB | 4.59 MiB/s, done.
Resolving deltas: 100% (885/885), completed with 255 local objects.
From https://github.com/NixOS/nixpkgs
   f102bcee7d1..3af210329f8  master                -> refs/nixpkgs-review/0
   4bd0c2ba1ca..f1292a953e5  refs/pull/122828/head -> refs/nixpkgs-review/1
$ git worktree add /home/siraben/.cache/nixpkgs-review/pr-122828-1/nixpkgs 3af210329f8dcbcaac23a9553a21acd5628a672f
Preparing worktree (detached HEAD 3af210329f8)
Updating files: 100% (27107/27107), done.
HEAD is now at 3af210329f8 Merge pull request #131151 from tomfitzhenry/patch-1
$ nix-env --option system x86_64-linux -f /home/siraben/.cache/nixpkgs-review/pr-122828-1/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff f1292a953e53205ea03b83bde847bf44304d0e6f
Auto-merging pkgs/top-level/all-packages.nix
Removing pkgs/applications/science/misc/openmodelica/update-src-libs-svn.sh
Removing pkgs/applications/science/misc/openmodelica/update-src-libs-git.sh
Removing pkgs/applications/science/misc/openmodelica/src-main.nix
Removing pkgs/applications/science/misc/openmodelica/src-libs-svn.nix
Removing pkgs/applications/science/misc/openmodelica/src-libs-git.nix
Removing pkgs/applications/science/misc/openmodelica/fakegit.nix
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/siraben/.cache/nixpkgs-review/pr-122828-1/nixpkgs -qaP --xml --out-path --show-trace --meta
1 package removed:
openmodelica (†)

Nothing to be built.

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/122828

Posting result comment on https://github.com/NixOS/nixpkgs/pull/122828
$ nix-shell /home/siraben/.cache/nixpkgs-review/pr-122828-1/shell.nix
```

**Comment by balodja** [comment]
> 1 package removed:
> openmodelica (†)
> 
> Nothing to be built.

Ah, I guess I see the problem. Should be corrected by now. e393ccf

File: pkgs/applications/science/misc/openmodelica/omparser/default.nix [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+  omdir = "OMParser";
+  omdeps = [ openmodelica.omcompiler ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ jre8 libuuid ];
+
+  patches = [ ./Makefile.in.patch ];
+
+  meta = with lib; {
```
Comment (siraben) [inline pkgs/applications/science/misc/openmodelica/omparser/default.nix:19]: You didn't import lib here

File: pkgs/applications/science/misc/openmodelica/omsimulator/default.nix [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+mkOpenModelicaDerivation rec {
+  pname = "omsimulator";
+  omdir = "OMSimulator";
+  omdeps = [ openmodelica.omcompiler ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ readline libxml2 boost ];
+
+  meta = with lib; {
```
Comment (siraben) [inline pkgs/applications/science/misc/openmodelica/omsimulator/default.nix:18]: lib is not imported

**Comment by siraben** [comment]
Also, after everything looks good, please squash commits when possible to have a clearer history.

File: pkgs/applications/science/misc/openmodelica/omparser/default.nix (hunk 22 above)
Comment (balodja) [reply pkgs/applications/science/misc/openmodelica/omparser/default.nix:20]: What a shame. I even didn't check the buildability after made changes.

Thanks! Corrected.

**Comment by balodja** [comment]
> Also, after everything looks good, please squash commits when possible to have a clearer history.

May I squash the whole PR to one commit? I don't think it's valuable to have all those changes in nixpkgs history.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 122828` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>openmodelica.combined</li>
    <li>openmodelica.omcompiler</li>
    <li>openmodelica.omedit</li>
    <li>openmodelica.omlibrary</li>
    <li>openmodelica.omparser</li>
    <li>openmodelica.omplot</li>
    <li>openmodelica.omshell</li>
    <li>openmodelica.omsimulator</li>
  </ul>
</details>

**Comment by siraben** [comment]
LGTM, please squash the commits then I'll merge.

**Review by siraben — APPROVED** [review]

**Comment by balodja** [comment]
@siraben , done.

**Comment by siraben** [comment]
Thanks again!

**Pull request merged**